source/opt/module.h

```diff
@@ -295,6 +306,13 @@ class Module {
   // If the module ends with Op*Line instruction, they will not be attached to
   // any instruction.  We record them here, so they will not be lost.
   std::vector<Instruction> trailing_dbg_line_info_;
+
+  // This module contains DebugScope or DebugNoScope.
+  bool contains_debug_scope_;
+
+  // A map between a result id of a local variable or a value of a local
+  // variable and its corresponding DebugDeclare or DebugValue.
+  std::unordered_map<uint32_t, std::unique_ptr<Instruction>> local_var_info_;
```

Comment:
Why do you need a map and not just a vector of instructions?

Reply:
We need to discuss this more.  The debug declare should be in the body of the function (in the basic block), and the debug value has to be the same.  I'll try  coming up with an example to demonstrate.